test/unit/BPool/BPool_JoinPool.tree

```diff
@@ -0,0 +1,23 @@
+BPool::JoinPool
+├── when reentrancy lock is set
+│   └── it should revert
+├── when pool is not finalized
+│   └── it should revert
+├── when total supply is zero
+│   └── it should revert
+├── when pool amount out is zero
+│   └── it should revert
+├── when pool ratio rounds down to zero
+│   └── it should revert
+├── when balance of pool in an underlying token is zero
```

Comment:
not following here, what's the condition and why it should revert?

Reply:
when the balance of any token is zero?
i'd use the "any token" also for the "when the required amount..."
and in the tests we just use 2nd token (avoid using the 1st one)